### PR TITLE
TMDM-14697 [CVE] - Update Log4J 1.x

### PR DESCRIPTION
**Pull request by VictorLi-talendbj**
https://jira.talendforge.org/browse/TMDM-14697

**What is the current behavior?** (You should also link to an open issue here)

Veracode is reporting a high level CVE against Log4J 1.2.17 as defined in mdm-common/org.talend.mdm.common/pom.xml

**What is the new behavior?**

Updated Log4j from 1.2.17 to 2.12.1, this version can also work with PowerMock/Junit.
And updated log4j.xml to log4j2 format, and changed log4j.properties to log4j2.xml for Junit.

**Please check if the PR fulfills these requirements**

- [X] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/query/optimization/ImplicitOrderBy.java [hunk 1]
```diff
@@ -12,16 +12,17 @@
...
+import org.apache.logging.log4j.LogManager;
 import org.talend.mdm.commmon.metadata.FieldMetadata;
 
 import java.util.HashSet;
 import java.util.Set;
 
 public class ImplicitOrderBy implements Optimizer {
 
-    private static final Logger               LOGGER                  = Logger.getLogger(ImplicitOrderBy.class);
+    private static final Logger               LOGGER                  = LogManager.getLogger(ImplicitOrderBy.class);
```
Comment (lpw82gf) [inline org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/query/optimization/ImplicitOrderBy.java:25]: Format the code.

File: org.talend.mdm.core/src/com/amalto/core/delegator/IValidation.java [hunk 2]
```diff
@@ -29,7 +30,7 @@
 
 public class IValidation {
 
-    private static final Logger LOGGER        = Logger.getLogger(IValidation.class);
+    private static final Logger LOGGER        = LogManager.getLogger(IValidation.class);
```
Comment (lpw82gf) [inline org.talend.mdm.core/src/com/amalto/core/delegator/IValidation.java:33]: Format the code

Comment (VictorLi-talendbj) [reply org.talend.mdm.core/src/com/amalto/core/delegator/IValidation.java]: Done

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/query/optimization/ImplicitOrderBy.java (hunk 1 above)
Comment (VictorLi-talendbj) [reply org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/query/optimization/ImplicitOrderBy.java]: Done

**Pull request merged**